python/ray/data/_internal/logical/operators/all_to_all_operator.py

```diff
@@ -125,7 +125,6 @@ def __init__(
             "Sort",
             input_op,
             sub_progress_bar_names=[
-                SortTaskSpec.SORT_SAMPLE_SUB_PROGRESS_BAR_NAME,
```

Comment:
Thanks for looking into this. I think we actually want to keep this bar here, so that it is initialized as a sub-progress bar for the `AbstractAllToAll` op, and we should remove the one that is created in `SortTaskSpec.sample_boundaries()`. We want the "Sort Sample" to be a sub-bar of the overall `Sort` bar, like this:
![357460721-30aa9fc3-8e96-473e-a794-da4fc023093a](https://github.com/user-attachments/assets/828646a5-751c-4bc8-982f-e0625b650815)

In terms of the concrete change, we can update `sample_boundaries()` to reference the sub-progress bar that is part of the operator. This will also require us to either:

(1) change `sample_boundaries` from a static method to class method, and use the object's sub-progress bar
(2) add a `sub_progress_bar` parameter to `sample_boundaries()`, and at the callers of this method, pass the `SortTaskSpec` instance's progress bar (e.g. [here](https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/planner/sort.py#L47))